library/core/src/error.rs

```diff
@@ -123,16 +123,20 @@ pub trait Error: Debug + Display {
 
     /// Provides type based access to context intended for error reports.
     ///
-    /// Used in conjunction with [`Demand::provide_value`] and [`Demand::provide_ref`] to extract
+    /// Used in conjunction with [`Request::provide_value`] and [`Request::provide_ref`] to extract
     /// references to member variables from `dyn Error` trait objects.
     ///
     /// # Example
     ///
     /// ```rust
-    /// #![feature(provide_any)]
     /// #![feature(error_generic_member_access)]
     /// use core::fmt;
-    /// use core::any::Demand;
+    /// use core::error::{request_ref, Request};
```

Comment:
Something is odd here, how does this doctest pass when it uses `core::error` without the `error_in_core` feature gate?

Reply:
E.g. if I press the "run" button [here](https://doc.rust-lang.org/nightly/std/error/trait.Error.html#method.provide), and then try to run the code in playground, I get an error.

https://github.com/rust-lang/rust/pull/114837 should fix that. But it is concerning that CI didn't catch this.